lib/practice_exercises.rb

```diff
@@ -1,13 +1,43 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: OofNsquared bc delete_at takes up ALOT of space (so says my tutor)
+# Space Complexity: Oof1 constant bc were not adding to it, we're subtracting
 def remove_duplicates(list)
-  raise NotImplementedError, "Not implemented yet"
+  i = 0
+  list.length.times do |num|
+    if list[i] == list [i + 1] #if this matches its neighbor...
+      list.delete_at(i +1)  #...delete it!
+    else
+      i += 1 #move on
+    end 
+  end
+  return list 
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+#hi chris, this one boggled my mind, dont mind my notes. 
+# Time Complexity: OofNsquared *maybe?
```

Comment:
That's a good idea, it's O(n * m) where n is the number of strings and m is the length of the strings.